### ISSUE TITLE
ABM with intervals in game_speed

### ISSUE THREAD

**Issue by kno10** [minetest/minetest#15293]
### Problem

For certain things such as water, having the ABM intervals in "real seconds" is fine.
But for other parts it might be desirable to rather use "game world milihours" instead.

For example, it may be desirable to have plants grow in days of the game world; and if a user sets the days to be twice as long, have them grow half as fast; but also keep the properties of random ticks. This of course causes a whole set of own play issues when the day cycle is set to a very short interval, or to zero; some users may want the plants to continue to grow and just disable the day-night cycle; while other use this for structure editing where they literally want to halt time (including plant growth and other game mechanics). It will need some experimentation how to have these best

### Solutions

- an optional new field `interval_unit` that can be set to `milihours` or `seconds`
- some API mechanism to skip game time, e.g., when sleeping, where only ABMs are run; possibly controlled by an optional `catch_up` setting of the ABM

### Alternatives

Use additional ABM ticks, skip unnecessary, and add complex catch-up logics when the ticks happen too infrequently.

**Comment by appgurueu** [comment]
Wouldn't it be possible to calculate the necessary ABM interval at load time based on `minetest.settings:get_float"time_speed"`? If so this issue would reduce to a comparatively low priority convenience feature and possibly a slightly different feature request to increase ABM granularity in my eyes.

**Comment by kno10** [comment]
The `time_speed` apparently can be changed during the game with `/set`, and I don't know if there is any way to change the registration of an ABM; or to get a callback when the setting is changed. But yes, I do not think this is a high priority.

**Comment by jeremyshannon** [comment]
Yes, changing the registration of ABMs after start is impossible. To keep the ABMs aligned with the time you'd have to restart.
